src/Strategy/R.hs

```diff
@@ -0,0 +1,142 @@
+
+data RProject
+  = Renv (Path Abs Dir) RDescriptionFile (Path Abs File)
+  | RenvLockOnly (Path Abs Dir) (Path Abs File)
+  | DescriptionOnly (Path Abs Dir) RDescriptionFile
+  deriving (Show, Eq, Ord, Generic)
+
+instance ToJSON RDescriptionFile
+instance ToJSON RProject
+
+instance AnalyzeProject RProject where
+  analyzeProject _ = getDeps
+  analyzeProject' _ = getDeps
+
+discover ::
+  ( Has ReadFS sig m
+  , Has Diagnostics sig m
+  , Has (Reader AllFilters) sig m
+  ) =>
+  Path Abs Dir ->
+  m [DiscoveredProject RProject]
+discover = simpleDiscover findProjects mkProject RProjectType
+
+findProjects ::
+  ( Has ReadFS sig m
+  , Has Diagnostics sig m
+  , Has (Reader AllFilters) sig m
+  ) =>
+  Path Abs Dir ->
+  m [RProject]
+findProjects = walkWithFilters' $ \dir _ files -> do
+  let lockFile = findFileNamed "renv.lock" files
+  let descriptionFile = findFileNamed "DESCRIPTION" files
+  let skipRenvLibs = WalkSkipSome ["renv"]
+
+  case (lockFile, descriptionFile) of
+    (Nothing, Just description') -> do
+      let project = DescriptionOnly dir (RDescriptionFile description')
+      pure ([project], skipRenvLibs)
+    (Just lockFile', Nothing) -> do
+      let project' = RenvLockOnly dir lockFile'
+      pure ([project'], skipRenvLibs)
+    (Just lockFile', Just description') -> do
+      let project' = Renv dir (RDescriptionFile description') lockFile'
+      pure ([project'], skipRenvLibs)
+    _ -> pure ([], WalkContinue)
+
+mkProject :: RProject -> DiscoveredProject RProject
+mkProject project =
+  DiscoveredProject
+    { projectType = RProjectType
+    , projectBuildTargets = mempty
+    , projectPath = projectDir project
+    , projectData = project
+    }
+  where
+    projectDir :: RProject -> Path Abs Dir
+    projectDir (Renv dir _ _) = dir
+    projectDir (RenvLockOnly dir _) = dir
+    projectDir (DescriptionOnly dir _) = dir
+
+getDeps :: (Has ReadFS sig m, Has Diagnostics sig m) => RProject -> m DependencyResults
+getDeps (Renv _ (RDescriptionFile descriptionFile) lockFile) = do
+  (graph, graphBreadth) <- analyzeDescriptionAndLockFile descriptionFile lockFile
+  pure $
+    DependencyResults
+      { dependencyGraph = graph
+      , dependencyGraphBreadth = graphBreadth
+      , dependencyManifestFiles = [lockFile]
+      }
+getDeps (RenvLockOnly _ lockFile) = do
+  void
+    . recover
+    . errCtx MissingDescriptionFile
+    $ fatalText "Cannot distinguish between direct and deep dependencies."
+
+  (graph, graphBreadth) <- analyzeLockFile lockFile
+  pure $
+    DependencyResults
+      { dependencyGraph = graph
+      , dependencyGraphBreadth = graphBreadth
+      , dependencyManifestFiles = [lockFile]
+      }
+getDeps (DescriptionOnly _ (RDescriptionFile descriptionFile)) = do
+  void
+    . recover
+    . warnOnErr MissingEdges
+    . warnOnErr MissingDeepDeps
+    . warnOnErr VersionConstraintsIgnored
+    . errCtx MissingRenvLockFile
+    $ fatalText "renv.lock file is missing."
```

Comment:
Previous errCtx and WarnOnErr will unfold, so it will look something like this in practice:

<img width="503" alt="CleanShot 2022-10-13 at 09 21 38@2x" src="https://user-images.githubusercontent.com/86321858/195638261-1a89f965-e9e5-46d0-8ba9-a965cd296628.png">